File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/adapters/viewholders/FeedbackViewHolder.java
package org.fossasia.susi.ai.skills.skilldetails.adapters.viewholders;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import org.fossasia.susi.ai.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by arundhati24 on 27/06/2018
 */

public class FeedbackViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    @BindView(R.id.itemFeedback)
    public LinearLayout itemFeedback;

    @BindView(R.id.tvInitials)
    public TextView initials;

    @BindView(R.id.tvEmail)
    public TextView feedbackEmail;

    @BindView(R.id.tvDate)
    public TextView feedbackDate;

    @BindView(R.id.tvFeedback)
    public TextView feedback;

    @BindView(R.id.seeAllReviews)
    public LinearLayout seeAllReviews;

    private FeedbackViewHolder.ClickListener listener;

    public FeedbackViewHolder(View itemView, FeedbackViewHolder.ClickListener clickListener) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        this.listener = clickListener;
        itemView.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        if (listener != null) {
            int position = getAdapterPosition();
            if (position == 3) {
                listener.onItemClicked(position);
            }
        }
    }

    public interface ClickListener {
        void onItemClicked(int position);
    }
}

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/adapters/viewholders/FeedbackViewHolder.kt
package org.fossasia.susi.ai.skills.skilldetails.adapters.viewholders

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import butterknife.ButterKnife
import kotterknife.bindView
import org.fossasia.susi.ai.R

/**
 * ViewHolder for drawing feed back item layout.
 */
class FeedbackViewHolder(
    itemView: View,
    private val listener: FeedbackViewHolder.ClickListener?
) :
    RecyclerView.ViewHolder(itemView), View.OnClickListener {

    val itemFeedback: LinearLayout by bindView(R.id.itemFeedback)
    val avatar: ImageView by bindView(R.id.avatar)
    val feedbackEmail: TextView by bindView(R.id.tvEmail)
    val feedbackDate: TextView by bindView(R.id.tvDate)
    val feedback: TextView by bindView(R.id.tvFeedback)
    val seeAllReviews: LinearLayout by bindView(R.id.seeAllReviews)

    init {
        ButterKnife.bind(this, itemView)
        itemView.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        if (listener != null) {
            val position = adapterPosition
            if (position == 3) {
                listener.onItemClicked(position)
            }
        }
    }

    interface ClickListener {
        fun onItemClicked(position: Int)
    }
}
